src/entities/Report/routes.ts

```diff
@@ -36,21 +38,34 @@ export async function getSignedUrl(
 
   const signedUrlExpireSeconds = 60 * 1000
 
-  const signedUrl = s3.getSignedUrl("putObject", {
-    Bucket: BUCKET_NAME,
-    Key: `${filename}`,
-    Expires: signedUrlExpireSeconds,
-    ContentType: mimetype,
-    ACL: "private",
-    CacheControl: "public, max-age=31536000, immutable",
-    Metadata: {
-      ...userAuth.metadata,
-      address: userAuth.address,
-    },
+  const signedUrl = await retry({}, async () => {
+    const responseUrl = s3.getSignedUrl("putObject", {
+      Bucket: BUCKET_NAME,
+      Key: `${filename}`,
+      Expires: signedUrlExpireSeconds,
+      ContentType: mimetype,
+      ACL: "private",
+      CacheControl: "public, max-age=31536000, immutable",
+      Metadata: {
+        ...userAuth.metadata,
+        address: userAuth.address,
+      },
+    })
+
+    const url = new URL(responseUrl)
+    if (url.searchParams.size === 0) {
+      throw new Error("Invalid AWS response")
+    }
+
+    return responseUrl
   })
 
+  const topLevelDomain = ENV === "prod" ? ".org" : ".zone"
+
+  const url = new URL(signedUrl)
+  url.hostname = `${BUCKET_NAME}.decentraland${topLevelDomain}`
```

Comment:
Better use an optional env `BUCKET_HOSTNAME` to replace the entire host instead of hardcoding decentraland